static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -352,7 +368,8 @@
           break
 
         default:
-          if (this.trigger_selection(e)) {
+          if (this.custom_selection_state.should_trigger) {
+            this.custom_selection_state.should_trigger = false;
```

Comment:
This code is kinda hard to follow.  Are we confident that `this.custom_selection_state` is always set during a `keyup` call?  And is the `keyCode` in `custom_selection_state` used for anything?

Reply:
Yes, a keyup event implies that we have already had a keypress event.